### PR TITLE
FOGL-5099: addition of purge option

### PR DESCRIPTION
**Pull request by pintomax**
FOGL-5099: addition of purge option

Initial implementation with Sqlite3 storage engine

### CODE REVIEW

**Comment by pintomax** [comment]
Next commits will be for Postgres storage engine

**Comment by pintomax** [comment]
Note: Postgres indexes are not reset right now

**Review by sstefano69 — COMMENTED** [review]
looks good to me 
I have done some tests, with Massimiliano also, and they succeed

File: scripts/plugins/storage/sqlite.sh [hunk 1]
```diff
@@ -426,15 +426,70 @@ Arguments:
...
+
+    # Log success
+    if [[ "$1" == "noisy" ]]; then
+        sqlite_log "info" "Purge complete for Fledge Plugin '${PLUGIN} in database '${DEFAULT_SQLITE_DB_FILE}'." "all" "pretty"
+    else
+        sqlite_log "info" "Purge complete for Fledge Plugin '${PLUGIN} in database '${DEFAULT_SQLITE_DB_FILE}'." "logonly" "pretty"
+    fi
+
+    # Remove all readings
+    sqlite_reset_db_readings
```
[the remark this answers is not in the document]
Comment (pintomax) [reply scripts/plugins/storage/sqlite.sh:491]: sqlite_reset takes $1 log verbosity, $2 is action: if "immediate" no user intervention is needed

"immediate" is being passed at database creation time

[the remark this answers is not in the document]
Comment (pintomax) [reply scripts/plugins/storage/sqlite.sh:491]: @praveen-garg  we did check for no readings dbs left after purge, checking it again

[the remark this answers is not in the document]
Comment (pintomax) [reply scripts/plugins/storage/sqlite.sh:491]: @praveen-garg after purge only data/readings_1.db database exists and it has readings_1_1 table which is empty.

Please confirm the table is empty:
sqlite> select count(1) from readings_1_1;
0

[the remark this answers is not in the document]
Comment (praveen-garg) [reply scripts/plugins/storage/sqlite.sh:491]:
```
root@7053d74eaa4c:/usr/local/fledge# sqlite3 data/readings_1.db
SQLite version 3.22.0 2018-01-22 18:45:57
Enter ".help" for usage hints.
sqlite> select count(*) from readings_1_1;
28
sqlite> .quit
root@7053d74eaa4c:/usr/local/fledge# ./bin/fledge purge
This script will remove all readings data and non-configuration data stored in the SQLite3 datafiles:
'/usr/local/fledge/data/fledge.db'
'/usr/local/fledge/data/readings_1.db'
Enter YES if you want to continue: YES
root@7053d74eaa4c:/usr/local/fledge# sqlite3 data/readings_1.db
SQLite version 3.22.0 2018-01-22 18:45:57
Enter ".help" for usage hints.
sqlite> select count(*) from readings_1_1;
28
```

using `make install`

[the remark this answers is not in the document]
Comment (praveen-garg) [reply scripts/plugins/storage/sqlite.sh:491]: **reset**

```
root@7053d74eaa4c:/usr/local/fledge# ./bin/fledge reset
This script will remove all data stored in the SQLite3 datafiles:
'/usr/local/fledge/data/fledge.db'
'/usr/local/fledge/data/readings_1.db'
Enter YES if you want to continue: YES
Removed user data from /usr/local/fledge/data/scripts
Removed user data from /usr/local/fledge/data/logs
Removed user data from /usr/local/fledge/data/var
Removed user data from /usr/local/fledge/data/extras
Removed user data from /usr/local/fledge/data/etc
Removed core.err from /usr/local/fledge/data
Removed user data from /usr/local/fledge/data/snapshots
Removed user data from /usr/local/fledge/data/backup
Removed user data from /usr/local/fledge/data/support
Removed user CSV files from /usr/local/fledge/data
root@7053d74eaa4c:/usr/local/fledge# sqlite3 data/readings_1.db
SQLite version 3.22.0 2018-01-22 18:45:57
Enter ".help" for usage hints.
sqlite> select count(*) from readings_1_1;
0
```

[the remark this answers is not in the document]
Comment (praveen-garg) [reply scripts/plugins/storage/sqlite.sh:491]: > re: question: what is the use of $2 being passed in

I meant to write for, 

```
    sqlite_reset_db_fledge   "$1" "$2"
    sqlite_reset_db_readings "$1" "$2"
```

[the remark this answers is not in the document]
Comment (pintomax) [reply scripts/plugins/storage/sqlite.sh:491]: > ```
> root@7053d74eaa4c:/usr/local/fledge# sqlite3 data/readings_1.db
> SQLite version 3.22.0 2018-01-22 18:45:57
> Enter ".help" for usage hints.
> sqlite> select count(*) from readings_1_1;
> 28
> sqlite> .quit
> root@7053d74eaa4c:/usr/local/fledge# ./bin/fledge purge
> This script will remove all readings data and non-configuration data stored in the SQLite3 datafiles:
> '/usr/local/fledge/data/fledge.db'
> '/usr/local/fledge/data/readings_1.db'
> Enter YES if you want to continue: YES
> root@7053d74eaa4c:/usr/local/fledge# sqlite3 data/readings_1.db
> SQLite version 3.22.0 2018-01-22 18:45:57
> Enter ".help" for usage hints.
> sqlite> select count(*) from readings_1_1;
> 28
> ```
> 
> using `make install`

@praveen-garg now fixed.

[the remark this answers is not in the document]
Comment (praveen-garg) [reply scripts/plugins/storage/sqlite.sh:491]: This works now! thanks.

**Comment by praveen-garg** [comment]
could you please check these as well?

packages, 
omf_created_objects, 
plugin_data, 
user_logins

**Review by sstefano69 — COMMENTED** [review]
new test with the latest code, it worked fine

**Review by MarkRiddoch — APPROVED** [review]

**Pull request merged**